### PR TITLE
URL builder cleanup

### PR DESCRIPTION
**Pull request by nsherred**
- refactor url building steps into their own functions
- handle url query using a dom event handler
- isolate dom interactions from url building

### CODE REVIEW

File: lawsearch.html [hunk 1]
```diff
@@ -164,52 +164,104 @@ <h1 class="pagetitle">Law&nbsp;Search</h1>
...
-if successful, false otherwise. Outputs error messages to the
-DOM element with id "explainer".
+  Parses URL search query, and passes it to handleQuery().
+  Runs upon page load, while the LawSearch page is hidden.
+  Returns true if the page should be unhidden, i.e. upon a *failed*
+  search. Returns false otherwise.
+  Also puts the query into the search bar.
 */
-function lookup(query) {
+document.addEventListener("DOMContentLoaded", () => {
```
Comment (nsherred) [inline lawsearch.html:176]: I moved the onload url query parsing/building out of the html, this allows us to more control over when the onload handler is called.

File: lawsearch.html [hunk 2]
```diff
@@ -164,52 +164,104 @@ <h1 class="pagetitle">Law&nbsp;Search</h1>
...
+  handleQuery(query);
+});
+
+function handleSearch(event) {
+  event.preventDefault()
+  const query = document.getElementById("q").value
+  handleQuery(query);
+}
+
+function handleQuery(query) {
```
Comment (nsherred) [inline lawsearch.html:193]: Dom interactions were pulled into the top level functions handleSearch, handleQuery, urlQuery to isolate the url building code from the dom interactions. This separates our concerns.

**Comment by nsherred** [comment]
These are just some recommendations, you definitely don't actually have to merge this pr. The pr format just allows me to add comments about specific lines of code.

File: lawsearch.html (hunk 1 above)
Comment (raindrum) [reply lawsearch.html:176]: That makes so much more sense! Honestly I didn't even know about event listeners. Thanks!

File: lawsearch.html [hunk 3]
```diff
@@ -164,52 +164,104 @@ <h1 class="pagetitle">Law&nbsp;Search</h1>
...
+  try {
+    if (!query) return document.getElementById("explainer").innerHTML = "";
+    window.location.href = getUrlForQuery(query);
+  } catch (error) {
```
Comment (raindrum) [inline lawsearch.html:197]: I _knew_ was something I was missing about how to do error messages!

**Review by raindrum — APPROVED** [review]
This is brilliant! Thank you so much! I'm pretty new to JavaScript (and to good coding practices), so your edits have taught me a lot.

This HTML is actually generated from [some markdown](https://raw.githubusercontent.com/raindrum/raindrum.github.io/master/lawsearch.md), for reasons, so it'll technically be overwritten in the next build, but I'll transfer your edits into the source file first. Merging this PR in the meantime.

**Pull request merged**